File: evals/tabletest/eval-27-convert-from-testng/project/src/test/java/com/example/ShippingCostCalculatorTest.java
package com.example;

import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

import java.math.BigDecimal;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import static org.testng.Assert.assertEquals;

public class ShippingCostCalculatorTest {

    private ShippingCostCalculator calculator = new ShippingCostCalculator();

    @DataProvider(name = "shippingData")
    public Object[][] shippingData() {
        return new Object[][] {
            { "EU standard light",
              new ShippingZone("EU", "standard"), 0.5,
              new int[]{20, 15, 10}, false, null, null,
              Carrier.DHL, new BigDecimal("5.00") },
            { "EU standard medium",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{30, 20, 15}, false, null, null,
              Carrier.DHL, new BigDecimal("7.50") },
            { "EU standard heavy",
              new ShippingZone("EU", "standard"), 10.0,
              new int[]{40, 30, 20}, false, null, null,
              Carrier.DHL, new BigDecimal("12.50") },
            { "EU standard very heavy",
              new ShippingZone("EU", "standard"), 25.0,
              new int[]{50, 40, 30}, false, null, null,
              Carrier.UPS, new BigDecimal("20.00") },
            { "EU express light",
              new ShippingZone("EU", "express"), 0.5,
              new int[]{20, 15, 10}, false, null, null,
              Carrier.DHL, new BigDecimal("8.00") },
            { "EU express medium",
              new ShippingZone("EU", "express"), 3.0,
              new int[]{30, 20, 15}, false, null, null,
              Carrier.UPS, new BigDecimal("12.00") },
            { "US standard light",
              new ShippingZone("US", "standard"), 0.5,
              new int[]{20, 15, 10}, false, null, null,
              Carrier.FEDEX, new BigDecimal("7.00") },
            { "US express heavy",
              new ShippingZone("US", "express"), 10.0,
              new int[]{40, 30, 20}, false, null, null,
              Carrier.DHL, new BigDecimal("30.00") },
            { "dimensional weight override",
              new ShippingZone("EU", "standard"), 1.0,
              new int[]{70, 50, 10}, false, null, null,
              Carrier.DHL, new BigDecimal("12.50") },
            { "fragile surcharge",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{30, 20, 15}, true, null, null,
              Carrier.DHL, new BigDecimal("8.625") },
            { "insured surcharge",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{30, 20, 15}, false, new BigDecimal("500"), null,
              Carrier.DHL, new BigDecimal("10.50") },
            { "fragile and insured",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{30, 20, 15}, true, new BigDecimal("200"), null,
              Carrier.DHL, new BigDecimal("11.625") },
            { "hazmat handling",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{30, 20, 15}, false, null, "hazmat",
              Carrier.DHL, new BigDecimal("15.50") },
            { "oversize surcharge",
              new ShippingZone("EU", "standard"), 3.0,
              new int[]{120, 5, 5}, false, null, null,
              Carrier.DHL, new BigDecimal("17.50") },
            { "same rate DHL",
              new ShippingZone("EU", "express"), 3.0,
              new int[]{30, 20, 15}, false, null, null,
              Carrier.DHL, new BigDecimal("12.00") },
            { "same rate UPS",
              new ShippingZone("EU", "express"), 3.0,
              new int[]{30, 20, 15}, false, null, null,
              Carrier.UPS, new BigDecimal("12.00") },
            { "same rate FEDEX",
              new ShippingZone("EU", "express"), 3.0,
              new int[]{30, 20, 15}, false, null, null,
              Carrier.FEDEX, new BigDecimal("12.00") },
        };
    }

    @Test(dataProvider = "shippingData")
    public void testShippingCost(String scenario, ShippingZone zone,
            double weight, int[] dimensions, boolean fragile,
            BigDecimal insuredValue, String handling,
            Carrier carrier, BigDecimal expected) {
        List<Integer> dims = Arrays.stream(dimensions).boxed().collect(Collectors.toList());
        PackageOptions opts = new PackageOptions();
        if (fragile) opts.setFragile(true);
        if (insuredValue != null) opts.setInsuredValue(insuredValue);
        if (handling != null) opts.setHandling(handling);
        BigDecimal result = calculator.calculateShippingCost(zone, weight, dims, opts, carrier);
        assertEquals(result.compareTo(expected), 0, "Failed for: " + scenario);
    }
}

File: evals/tabletest/eval-26-convert-from-kotest/project/src/test/kotlin/com/example/ShippingCostCalculatorTest.kt
package com.example

import io.kotest.core.spec.style.FunSpec
import io.kotest.datatest.withData
import io.kotest.matchers.shouldBe
import java.math.BigDecimal

data class ShippingScenario(
    val name: String,
    val zone: ShippingZone,
    val weight: Double,
    val dimensions: List<Int>,
    val fragile: Boolean,
    val insuredValue: BigDecimal?,
    val handling: String?,
    val carrier: Carrier,
    val expected: BigDecimal
) {
    override fun toString() = name
}

class ShippingCostCalculatorTest : FunSpec({

    val calculator = ShippingCostCalculator()

    context("shipping cost calculation") {
        withData(
            ShippingScenario("EU standard light", ShippingZone("EU", "standard"), 0.5, listOf(20, 15, 10), false, null, null, Carrier.DHL, BigDecimal("5.00")),
            ShippingScenario("EU standard medium", ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), false, null, null, Carrier.DHL, BigDecimal("7.50")),
            ShippingScenario("EU standard heavy", ShippingZone("EU", "standard"), 10.0, listOf(40, 30, 20), false, null, null, Carrier.DHL, BigDecimal("12.50")),
            ShippingScenario("EU standard very heavy", ShippingZone("EU", "standard"), 25.0, listOf(50, 40, 30), false, null, null, Carrier.UPS, BigDecimal("20.00")),
            ShippingScenario("EU express light", ShippingZone("EU", "express"), 0.5, listOf(20, 15, 10), false, null, null, Carrier.DHL, BigDecimal("8.00")),
            ShippingScenario("EU express medium", ShippingZone("EU", "express"), 3.0, listOf(30, 20, 15), false, null, null, Carrier.UPS, BigDecimal("12.00")),
            ShippingScenario("US standard light", ShippingZone("US", "standard"), 0.5, listOf(20, 15, 10), false, null, null, Carrier.FEDEX, BigDecimal("7.00")),
            ShippingScenario("US express heavy", ShippingZone("US", "express"), 10.0, listOf(40, 30, 20), false, null, null, Carrier.DHL, BigDecimal("30.00")),
            ShippingScenario("dimensional weight override", ShippingZone("EU", "standard"), 1.0, listOf(70, 50, 10), false, null, null, Carrier.DHL, BigDecimal("12.50")),
            ShippingScenario("fragile surcharge", ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), true, null, null, Carrier.DHL, BigDecimal("8.625")),
            ShippingScenario("insured surcharge", ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), false, BigDecimal("500"), null, Carrier.DHL, BigDecimal("10.50")),
            ShippingScenario("fragile and insured", ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), true, BigDecimal("200"), null, Carrier.DHL, BigDecimal("11.625")),
            ShippingScenario("hazmat handling", ShippingZone("EU", "standard"), 3.0, listOf(30, 20, 15), false, null, "hazmat", Carrier.DHL, BigDecimal("15.50")),
            ShippingScenario("oversize surcharge", ShippingZone("EU", "standard"), 3.0, listOf(120, 5, 5), false, null, null, Carrier.DHL, BigDecimal("17.50")),
            ShippingScenario("same rate DHL", ShippingZone("EU", "express"), 3.0, listOf(30, 20, 15), false, null, null, Carrier.DHL, BigDecimal("12.00")),
            ShippingScenario("same rate UPS", ShippingZone("EU", "express"), 3.0, listOf(30, 20, 15), false, null, null, Carrier.UPS, BigDecimal("12.00")),
            ShippingScenario("same rate FEDEX", ShippingZone("EU", "express"), 3.0, listOf(30, 20, 15), false, null, null, Carrier.FEDEX, BigDecimal("12.00"))
        ) { scenario ->
            val opts = PackageOptions()
            if (scenario.fragile) opts.isFragile = true
            if (scenario.insuredValue != null) opts.insuredValue = scenario.insuredValue
            if (scenario.handling != null) opts.handling = scenario.handling
            calculator.calculateShippingCost(scenario.zone, scenario.weight, scenario.dimensions, opts, scenario.carrier).compareTo(scenario.expected) shouldBe 0
        }
    }
})
